pulp_rpm/tests/functional/api/test_acs.py

```diff
@@ -19,94 +9,81 @@
-        cls.acs_api = AcsRpmApi(cls.client)
-        cls.repo_api = RepositoriesRpmApi(cls.client)
-        cls.remote_api = RemotesRpmApi(cls.client)
-
-        delete_orphans()
-
-    @classmethod
-    def tearDownClass(cls):
-        """Tear down."""
-        delete_orphans()
-
-    def do_test(self, acs_url, paths, remote_url):
+@pytest.fixture
+def acs_test_factory(
+    rpm_repository_api,
+    rpm_acs_api,
+    rpm_repository_factory,
+    rpm_rpmremote_factory,
+    monitor_task,
+    monitor_task_group,
+    gen_object_with_cleanup,
+    delete_orphans_pre,
+):
+    def _do_test(acs_url, paths, remote_url):
         """Sync with ACS test."""
         # ACS is rpm-unsigned repository which has all packages needed
-        acs_remote = self.remote_api.create(gen_rpm_remote(url=acs_url, policy="on_demand"))
-        self.addCleanup(self.remote_api.delete, acs_remote.pulp_href)
+        acs_remote = rpm_rpmremote_factory(url=acs_url, policy="on_demand")
 
         acs_data = {
             "name": "alternatecontentsource",
             "remote": acs_remote.pulp_href,
             "paths": paths,
         }
-        acs = self.acs_api.create(acs_data)
-        self.addCleanup(self.acs_api.delete, acs.pulp_href)
+        acs = gen_object_with_cleanup(rpm_acs_api, acs_data)
 
-        repo = self.repo_api.create(gen_repo())
-        self.addCleanup(self.repo_api.delete, repo.pulp_href)
-        remote = self.remote_api.create(gen_rpm_remote(url=remote_url))
-        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+        repo = rpm_repository_factory()
+        remote = rpm_rpmremote_factory(url=remote_url)
 
         # Sync repo with metadata only, before ACS refresh it should fail
         repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
 
-        with self.assertRaises(PulpTaskError) as ctx:
-            sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        with pytest.raises(PulpTaskError) as ctx:
+            sync_response = rpm_repository_api.sync(repo.pulp_href, repository_sync_data)
             monitor_task(sync_response.task)
 
-        self.assertIn("404, message='Not Found'", ctx.exception.task.error["description"])
+        assert "404, message='Not Found'" in ctx.value.task.error["description"]
 
         # ACS refresh
-        acs_refresh = self.acs_api.refresh(acs.pulp_href)
+        acs_refresh = rpm_acs_api.refresh(acs.pulp_href)
         monitor_task_group(acs_refresh.task_group)
 
         # Sync repository with metadata only
-        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        sync_response = rpm_repository_api.sync(repo.pulp_href, repository_sync_data)
         monitor_task(sync_response.task)
 
-        return self.repo_api.read(repo.pulp_href)
+        return rpm_repository_api.read(repo.pulp_href)
+
+    return _do_test
+
+
+def test_acs_simple(acs_test_factory, rpm_repository_version_api):
```

Comment:
I think instead of `acs_test_factory` we could just parameterize these, right?  The only distinction appears to be the URL and the content summary that it checks against.

Reply:
Correct. I'll do that.